src/plugins/ui_core_views/sheetview.ts

```diff
@@ -180,20 +180,16 @@ export class SheetViewPlugin extends UIPlugin {
         this.setSheetViewOffset(cmd.offsetX, cmd.offsetY);
         break;
       case "SHIFT_VIEWPORT_DOWN":
-        const { top } = this.getActiveMainViewport();
         const sheetId = this.getters.getActiveSheetId();
-        const shiftedOffsetY = this.clipOffsetY(
-          this.getters.getRowDimensions(sheetId, top).start + this.sheetViewHeight
-        );
-        this.shiftVertically(shiftedOffsetY);
+        const { top, viewportHeight, offsetCorrectionY } = this.getMainInternalViewport(sheetId);
```

Comment:
Maybe this kind of changes (calling directly getMainInternalViewport instead of getActiveMainViewport) can be done in the line below to avoid multiple imbricated call and a zone reconstruction in `getMainViewport` :)

https://github.com/odoo/o-spreadsheet/pull/3981/files#diff-1954abb89fa17ccd5406b15bba7cf0a433c1a628a56cdde30d58074c6249d012L146